graf3d/eve7/inc/ROOT/REveElement.hxx

```diff
@@ -16,10 +16,22 @@
 #include <ROOT/REveVector.hxx>
 #include <ROOT/REveProjectionBases.hxx>
 
-#include <map>
 #include <memory>
+#include <list>
+#include <ostream>
+#include <set>
+#include <string>
+
+#if __has_include(<nlohmann/json_fwd.hpp>)
```

Comment:
Should one use it like:
```
#if defined __has_include && __has_include(<nlohmann/json_fwd.hpp>)
```

Reply:
See https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005finclude.html

Reply:
`__has_include` is part of C++17, so we don't need that: https://en.cppreference.com/w/cpp/preprocessor/include :tada:

Reply:
> Should one use it like:
> 
> ```
> #if defined __has_include && __has_include(<nlohmann/json_fwd.hpp>)
> ```

```
#ifdef __has_include
#if __has_include(<nlohmann/json_fwd.hpp>)
```

> See [gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005finclude.html](https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005finclude.html)

Exactly. Don't put those tests on the same line. Preprocessors that don't know about `__has_include` will error out on `<`.